fe/fe-core/src/main/java/org/apache/doris/nereids/types/VarcharType.java

```diff
@@ -38,4 +40,18 @@ public static VarcharType createVarcharType(int len) {
     public Type toCatalogDataType() {
         return ScalarType.createVarcharType(len);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!super.equals(o)) {
+            return false;
+        }
+        VarcharType that = (VarcharType) o;
```

Comment:
this may throw java.lang.ClassCastException. It is better to compare their class before convert.

Reply:
super.equals() include it.